random-forest/src/main/java/com/example/RandomForestApp.java

```diff
@@ -0,0 +1,119 @@
+        DataSource irisSource = new DataSource("iris.arff");
+        Instances irisData = irisSource.getDataSet();
+        if (irisData.classIndex() == -1) {
+           irisData.setClassIndex(irisData.numAttributes() - 1);
+        }
+
+        // Create a SigOpt experiment for the Random Forest parameters
+        Experiment experiment = Experiment.create(
+            new Experiment.Builder()
+                .name("Random Forest (Java)")
+                .parameters(Arrays.asList(
+                    new Parameter.Builder()
+                        .name("maxDepth")
+                        .type("int")
+                        .bounds(new Bounds(1.0, 50.0))
+                        .build(),
+                    new Parameter.Builder()
+                        .name("bagSizePercent")
+                        .type("int")
+                        .bounds(new Bounds(1.0, 100.0))
+                        .build(),
+                    new Parameter.Builder()
+                        .name("numIterations")
+                        .type("int")
+                        .bounds(new Bounds(1.0, 100.0))
+                        .build()
+                ))
+                .observationBudget(60)
+                .build())
+            .call();
+        System.out.println("Created experiment: https://sigopt.com/experiment/" + experiment.getId());
+
+        // Run the Optimization Loop between 10x - 20x the number of parameters
+        for (int i = 0; i < experiment.getObservationBudget(); i++) {
+            // Receive a Suggestion from SigOpt
+            Suggestion suggestion = experiment.suggestions().create().call();
+
+            // Evaluate the model locally
+            Result result = RandomForestApp.evaluateModel(suggestion.getAssignments(), irisData);
+
+            // Report an Observation (with standard deviation) back to SigOpt
+            Observation observation = experiment.observations().create()
+                .data(new Observation.Builder()
+                    .suggestion(suggestion.getId())
+                    .value(result.value)
+                    .valueStddev(result.stddev)
+                    .build())
+                .call();
+        }
+
+        // Re-fetch the experiment to get the best observed value and assignments
+        experiment = Experiment.fetch(experiment.getId()).call();
+        Assignments bestAssignments = experiment.getProgress().getBestObservation().getAssignments();
+        RandomForest bestRandomForest = new RandomForest();
+
+        // To wrap up the Experiment, fit the RandomForest on the best assigments and train on all available data
+        bestRandomForest.setMaxDepth(((Double)bestAssignments.get("maxDepth")).intValue());
+        bestRandomForest.setBagSizePercent(((Double)bestAssignments.get("bagSizePercent")).intValue());
+        bestRandomForest.setNumIterations(((Double)bestAssignments.get("numIterations")).intValue());
+        bestRandomForest.buildClassifier(irisData);
```

Comment:
It's a little different, here we are training the model with all of the available data, whereas in evaluateModel we're holding out data on every cross validation fold and testing.